### ISSUE TITLE
Handle JavaExec.main deprecation

### ISSUE THREAD

**Issue by borismorris-anaplan** [szpak/gradle-pitest-plugin#283]
The property `JavaExec.main` has been [deprecated][1] and is due for removal in Gradle 8.0. This property is used in the `PitestTask`.

The following stack trace is logged when running on Gradle 7.1.1

```
The JavaExec.main property has been deprecated. This is scheduled to be removed in Gradle 8.0. Please use the mainClass property instead. See https://docs.gradle.org/7.1.1/dsl/org.gradle.api.tasks.JavaExec.html#org.gradle.api.tasks.JavaExec:main for more details.
        at org.gradle.api.tasks.JavaExec.setMain(JavaExec.java:418)
        at info.solidsoft.gradle.pitest.PitestTask_Decorated.setMain(Unknown Source)
        at org.gradle.api.tasks.JavaExec.setMain(JavaExec.java:112)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at info.solidsoft.gradle.pitest.PitestTask_Decorated.setProperty(Unknown Source)
        at info.solidsoft.gradle.pitest.PitestTask.<init>(PitestTask.groovy:247)
        at info.solidsoft.gradle.pitest.PitestTask_Decorated.<init>(Unknown Source)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
        at org.gradle.internal.instantiation.generator.AsmBackedClassGenerator$InvokeConstructorStrategy.newInstance(AsmBackedClassGenerator.java:2070)
        at org.gradle.internal.instantiation.generator.AbstractClassGenerator$GeneratedClassImpl$GeneratedConstructorImpl.newInstance(AbstractClassGenerator.java:486)
       ... etc
```

[1]: https://docs.gradle.org/7.1.1/dsl/org.gradle.api.tasks.JavaExec.html#org.gradle.api.tasks.JavaExec:main

**Comment by bondolo** [comment]
Proposed a fix in #289 that appears to work correctly.

**Comment by szpak** [comment]
Thanks @bondolo. I will be a part of "just being released" 1.7.0.
